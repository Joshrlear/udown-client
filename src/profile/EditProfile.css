/* form
    width: 500px */
input {
  width: 80%;
}

::-webkit-input-placeholder {
  color: rgba(0, 0, 0, 0.4);
  font-style: italic;
}

:-ms-input-placeholder {
  color: rgba(0, 0, 0, 0.4);
  font-style: italic;
}

::-ms-input-placeholder {
  color: rgba(0, 0, 0, 0.4);
  font-style: italic;
}

::placeholder {
  color: rgba(0, 0, 0, 0.4);
  font-style: italic;
}

.input_note_container {
  background-color: rgba(0, 0, 0, 0.2);
  padding: 10px;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}

.save_btn {
  display: block;
  position: relative;
  width: 50px;
  height: 30px;
  right: -75%;
  border-radius: 20px;
  text-align: center;
  padding: 7px 0px 0px 0px;
  margin: 0px;
  text-decoration: none;
  text-shadow: -1px 1px 1px rgba(0, 0, 0, 0.4);
  color: white;
  font-weight: 600;
  background-color: #6ab4ce;
  -webkit-box-shadow: 5px 5px 4px rgba(64, 166, 255, 0.3) inset, -3px -3px 5px rgba(145, 255, 112, 0.3) inset;
          box-shadow: 5px 5px 4px rgba(64, 166, 255, 0.3) inset, -3px -3px 5px rgba(145, 255, 112, 0.3) inset;
  -webkit-transition: -webkit-box-shadow 0.2s ease-out;
  transition: -webkit-box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out, -webkit-box-shadow 0.2s ease-out;
}

.save {
  display: inline-block;
  width: 70px;
  height: 30px;
  font-size: 16px;
}

.back_btn {
  display: block;
  position: relative;
  width: 50px;
  height: 30px;
  right: -75%;
  border-radius: 20px;
  text-align: center;
  padding: 7px 0px 0px 0px;
  margin: 0px;
  text-decoration: none;
  text-shadow: -1px 1px 1px rgba(0, 0, 0, 0.4);
  color: white;
  font-weight: 600;
  background-color: #6ab4ce;
  -webkit-box-shadow: 5px 5px 4px rgba(64, 166, 255, 0.3) inset, -3px -3px 5px rgba(145, 255, 112, 0.3) inset;
          box-shadow: 5px 5px 4px rgba(64, 166, 255, 0.3) inset, -3px -3px 5px rgba(145, 255, 112, 0.3) inset;
  -webkit-transition: -webkit-box-shadow 0.2s ease-out;
  transition: -webkit-box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out, -webkit-box-shadow 0.2s ease-out;
}

.back_btn:hover {
  -webkit-box-shadow: 0px 0px 100px rgba(158, 239, 70, 0.5) inset, 0px 0px 100px rgba(158, 239, 70, 0.5) inset;
          box-shadow: 0px 0px 100px rgba(158, 239, 70, 0.5) inset, 0px 0px 100px rgba(158, 239, 70, 0.5) inset;
}
/*# sourceMappingURL=EditProfile.css.map */