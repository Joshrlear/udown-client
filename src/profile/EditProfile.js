import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import config from '../config'
import UdownContext from '../UdownContext';
import functions from '../functions';
import fetches from '../fetches'
import './profile.scss'
import { thisExpression } from '@babel/types';

const { getProfileImage, getProfilePhone } = fetches.profileFetches
const authFunctions = functions.authFunctions

const imageHeight = Math.ceil(window.outerHeight * 0.35);
const imageWidth = Math.ceil(window.innerWidth);

export default class EditProfile extends Component {
    constructor(props) {
      super(props)
      this.imageUpload = React.createRef()
      this.phoneNumber = React.createRef()
      this.state = {
        user_id: '',
        profileImage: `https://via.placeholder.com/${imageWidth}x${imageHeight}`,
        phone: '',
        imageUpload: 'Upload Image',
      }
    }

    static contextType = UdownContext;
    
    componentDidMount() {

      this.setState({
        "user_id": localStorage.user_id
      })
    }

    componentDidUpdate() {
      const user_id = localStorage.user_id

      // get profile image
      const imageResult = Promise.resolve(getProfileImage(user_id, this.props))
      imageResult.then(value => {
        if (value) {
          const base64Image = value.image.image
          const image = `data:image/jpg;base64, ${base64Image}`
          /* document.getElementById('profile-image').src = `data:image/jpg;base64, ${image}` */
          value && (
            this.state.profileImage !== image  && (
              this.setState({
                profileImage: image
              })
            )
          )
          
        }
      })

      // get user phone_number
      const phoneResult = Promise.resolve(getProfilePhone(user_id, 'phone_number'))
      phoneResult.then(value => {
        if (value) {
          this.state.phone !== value.field && (
            this.setState({
              phone: value.field
            })
          )
        }
      })
    }

    handleSubmit = (e) => {
      e.preventDefault()
      const user_id = localStorage.user_id
      const imageUpload = this.imageUpload.current.files[0] ? this.imageUpload.current.files[0] : null
      const imageName = this.imageUpload.current.files[0] ? `${Date.now()}-${this.imageUpload.current.files[0].name}` : null
      const phoneNumber = this.phoneNumber.current.value ? this.phoneNumber.current.value : null

      const formData = new FormData()
      this.imageUpload.current.files[0] && formData.append('image', imageUpload, imageName)
      formData.append('phone', phoneNumber)
      console.log('LLLLLLLLLLLL',user_id)
      fetch(`${config.API_ENDPOINT}profile/${user_id}`, {
        method: 'POST',
        body: formData,
        headers: { "user_id": user_id }
        //credentials: 'include' 
      })
      .then(res => {
        if (!res.ok) {
          return res.json().then(err => {
            throw err
          })
        }
        else {
          console.log(res)
          return res
        }
      })
      .catch(err => {
        console.log(err)
      })
    }

    handleValue = () => {
      const imgName = this.imageUpload.current.value.split('fakepath\\')[1]
      this.setState({
        imageUpload: imgName
      })
    }

    render() {
        return (
          <div className="edit_profile_container">
            <div className="edit_profile">
                <div className="img_container">
                    <img id='profile-image' className="profile_image" src={this.state.profileImage} alt="User profile" />
                </div>
                <form ref='uploadForm' 
                  id='uploadForm' 
                  onSubmit={e => this.handleSubmit(e)}
                  method='post' 
                  encType="multipart/form-data">
                  <input 
                    ref={ this.imageUpload }
                    onChange={ this.handleValue } 
                    type="file" 
                    name="imageUpload"
                    id="imageUpload" />
                    <label htmlFor="imageUpload" className="imageUpload"><span>{ this.state.imageUpload || 'No Image Selected' }</span></label>
                  <label className="phone_label">Edit Phone:</label>
                  <input 
                    ref={ this.phoneNumber } 
                    className="phone" 
                    placeholder="no number yet..." 
                    defaultValue={ this.state.phone } 
                    disabled={ localStorage.username == "test" }
                  />
                  <div className="input_note_container">
                    <p className="input_note">   
                      <em>
                          Phone number is used to send<br/>
                          invites via text message.
                      </em>
                    </p>
                  </div>
                  <div className="button_rack">
                    {/* <a className="save_btn"> */}
                      <input className="save_btn" type='submit' value='Save' />
                    {/* </a> */}
                  </div>
                </form>
            </div>
          </div>
        )
    }
}