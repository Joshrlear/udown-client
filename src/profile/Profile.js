import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import config from '../config';
import UdownContext from '../UdownContext';
import functions from '../functions';
import fetches from '../fetches'
import EditProfile from './EditProfile'
import './profile.scss'

const { getProfileImage } = fetches.profileFetches
const authFunctions = functions.authFunctions

export default class Profile extends Component {
  constructor(props) {
    super(props)
    this.state = {
      user_id: ''
    }
  }

    static contextType = UdownContext;

    
    componentWillMount() {
      authFunctions.redirectIfLoggedIn(this.props, this.context.isLoggedIn)
    }
    

    componentDidMount() {
      const user_id = localStorage.user_id

      const result = Promise.resolve(getProfileImage(user_id, this.props))
      result.then(value => {
        if (value) {
          const image = value.image.image
          document.getElementById('profile-image').src = `data:image/jpg;base64, ${image}`
        }
      })
    }

    render() {

      const imageHeight = Math.ceil(window.outerHeight * 0.35);
      const imageWidth = Math.ceil(window.innerWidth);

      return (
          <div className="profile_container">
              <div className="profile">
                <div className="img_container">
                  <img id='profile-image' className="profile_image" src={`https://via.placeholder.com/${imageWidth}x${imageHeight}`} alt="User profile" />
                </div>
                <div className="profile_info">
                  <header className="profile_header">
                    <h3 className="profile_name">Name</h3>
                  </header>
                </div>
                <Link
                  to="/profileEdit"
                  className="edit_btn">
                  Edit
                </Link>
              </div>
          </div>
      )
    }
}