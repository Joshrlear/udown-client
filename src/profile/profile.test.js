import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';

import Profile from './Profile';
import EditProfile from './EditProfile';

configure({ adapter: new Adapter() })

describe('Components', () => {
    const mappedComponents = [Profile, EditProfile]
    mappedComponents.map(Component => {
        it(`renders Components without crashing`, () => {
            const div = document.createElement('div')
            ReactDOM.render(
            <BrowserRouter>
               <Component/>
            </BrowserRouter>, div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })

    it('renders Profile correctly', () => {
        const wrapper = shallow(<Profile/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders EditProfile correctly', () => {
        const wrapper = shallow(<EditProfile/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})