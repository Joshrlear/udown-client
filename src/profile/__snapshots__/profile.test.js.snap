// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components renders EditProfile correctly 1`] = `
<div
  className="edit_profile_container"
>
  <div
    className="edit_profile"
  >
    <div
      className="img_container"
    >
      <img
        alt="User profile"
        className="profile_image"
        id="profile-image"
        src="https://via.placeholder.com/1024x269"
      />
    </div>
    <form
      encType="multipart/form-data"
      id="uploadForm"
      method="post"
      onSubmit={[Function]}
    >
      <input
        id="imageUpload"
        name="imageUpload"
        onChange={[Function]}
        type="file"
      />
      <label
        className="imageUpload"
        htmlFor="imageUpload"
      >
        <span>
          Upload Image
        </span>
      </label>
      <label
        className="phone_label"
      >
        Edit Phone:
      </label>
      <input
        className="phone"
        defaultValue=""
        placeholder="no number yet..."
      />
      <div
        className="input_note_container"
      >
        <p
          className="input_note"
        >
          <em>
            Phone number is used to send
            <br />
            invites via text message.
          </em>
        </p>
      </div>
      <div
        className="button_rack"
      >
        <input
          className="save_btn"
          type="submit"
          value="Save"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`Components renders Profile correctly 1`] = `
<div
  className="profile_container"
>
  <div
    className="profile"
  >
    <div
      className="img_container"
    >
      <img
        alt="User profile"
        className="profile_image"
        id="profile-image"
        src="https://via.placeholder.com/1024x269"
      />
    </div>
    <div
      className="profile_info"
    >
      <header
        className="profile_header"
      >
        <h3
          className="profile_name"
        >
          Name
        </h3>
      </header>
    </div>
    <Link
      className="edit_btn"
      to="/profileEdit"
    >
      Edit
    </Link>
  </div>
</div>
`;
